packages/cyberstorm/src/components/componentStyles/Button.module.css

```diff
@@ -0,0 +1,145 @@
+  align-items: center;
+  justify-content: center;
+
+  padding: var(--padding-s) var(--padding-m);
+  border: var(--button-border-width) solid var(--button-border-color);
+  border-radius: var(--size-xs);
+  color: var(--text-color);
+  background-color: var(--bg-color);
+
+  --text-color: var(--color-white);
+  --bg-color: var(--color-surface--3);
+  --button-border-color: transparent;
+  --button-border-width: 2px;
+}
+
+.buttonLabel {
+  font-weight: 600;
+  font-size: var(--size-m);
+  font-family: Sora, sans-serif;
+  font-style: normal;
+  line-height: 1.125rem;
+}
+
+.button__default {
+  --text-color: var(--color-white);
+  --bg-color: var(--color-surface--3);
+}
+
+.button__default:hover {
+  --bg-color: var(--color-surface--4);
+  --button-border-color: var(--color-border-2);
+}
+
+.button__defaultDark {
+  --text-color: var(--color-white);
+  --bg-color: var(--color-surface--1);
+}
+
+.button__defaultDark:hover {
+  --bg-color: var(--color-surface--2);
+  --button-border-color: var(--color-border-2);
+}
+
+.button__primary {
+  --text-color: var(--color-black);
+  --bg-color: var(--color-cyber-green);
+}
+
+.button__primary:hover {
+  --bg-color: var(--color-cyber-green--40);
+}
+
+.button__danger {
+  --text-color: var(--color-white);
+  --bg-color: var(--color-red);
+}
+
+.button__danger:hover {
+  --bg-color: var(--color-red--40);
+}
+
+.button__defaultWithBorder {
+  --bg-color: var(--color-surface--3);
+  --text-color: var(--color-white);
+  --button-border-color: var(--color-border-2);
+}
+
+.button__defaultWithBorder:hover {
+  --bg-color: var(--color-surface--4);
+  --button-border-color: var(--color-border-3);
+}
+
+.button__specialGreen {
+  position: relative;
+  z-index: 1;
+  background: var(--color-gradient-blue-green--darker) padding-box, var(--color-gradient-blue-green) border-box;
+  box-shadow: 0 0 var(--size-m) rgb(80 217 159 / 0.32);
+  transition: all 0.12s;
+}
+
+.button__specialGreen::before {
+  position: absolute;
+  top: calc(0px - var(--button-border-width));
+  right: calc(0px - var(--button-border-width));
+  bottom: calc(0px - var(--button-border-width));
+  left: calc(0px - var(--button-border-width));
+  z-index: -1;
+  border: var(--button-border-width) solid transparent;
+  border-radius: var(--size-xs);
+  background: var(--color-gradient-blue-green--dark) padding-box, var(--color-gradient-blue-green--light) border-box;
+  opacity: 0;
+  transition: all 0.12s;
+  content: "";
+
+  --text-color: var(--color-white);
+}
+
+.button__specialGreen:hover {
+  --button-border-color: transparent;
+
+  box-shadow: 0 0 var(--size-xs) rgb(255 255 255 / 0.2), 0 0 var(--size-l) rgb(80 217 159 / 0.6);
+}
+
+.button__specialGreen:hover::before {
+  opacity: 1;
+}
+
+.button__specialPurple {
+  position: relative;
+  z-index: 1;
+  background: var(--color-gradient-purple-green--darker) padding-box, var(--color-gradient-purple-green) border-box;
+  transition: all 0.12s;
+}
+
+.button__specialPurple::before {
```

Comment:
Just a random thought here, not requiring any action now. This has a lot of repetition with `.button__specialGreen::before` which would be nice to avoid somehow, while still hiding it from the Button component (i.e. not requiring it to use modifier classes etc. for `.button__specialFoo`.

Reply:
At the very least the two pseudo elements could share the rules sans `background`, as that's the only one that differs.

Reply:
Refactored, should be at least better now